src/coreclr/jit/ssabuilder.cpp

```diff
@@ -1311,7 +1298,7 @@ void SsaBuilder::AddPhiArgsToSuccessors(BasicBlock* block)
                     GenTreePhi* phi    = tree->AsLclVar()->Data()->AsPhi();
                     unsigned    ssaNum = m_renameStack.Top(lclNum);
 
-                    AddPhiArg(handlerStart, stmt, phi, lclNum, ssaNum, block);
+                    AddPhiArg(handlerStart, stmt, phi, lclNum, ssaNum, succ);
```

Comment:
This seemed a bit questionable before -- e.g. for
```csharp
x = 123;
try
{
  M();
  x = 456;
}
catch
{
}
```
we would have no phi arg in the handler indicating that the value of `x` could be `123` even if control came from the `try` block.

Reply:
So this changes the catch from something like $x_2 = \phi ([x_0, pretry], [x_1, try])$ to $x_2 = \phi ([x_0, pretry], [x_0, try], [x_1, try])$?

Reply:
We don't have $pretry$ in the list anymore, so rather to $x_2 = \phi ([x_0, try], [x_1, try])$